### PR TITLE
tests: Don't overwrite CI statuses on main for nightly pipelines

### LINKED ISSUE #1527: CI: confusing failure - wrong version built?
I probably didn't investigate this enough but I wanted to note it down to have a better look tomorrow.

There's something odd about the current CI failure in main ([CI pipeline for 6e22e1de64c18299c80e13a7b9a60049b16ea040](https://gitlab.com/osbuild/ci/osbuild-composer/-/pipelines/331383627)).

The API test with argument `aws.s3` failed with the message: 
```
+ echo 'Unknown cloud provider '\''aws.s3'\''. Supported are '\''aws'\'', '\''gcp'\'''
Unknown cloud provider 'aws.s3'. Supported are 'aws', 'gcp'
```
[Link to job](https://gitlab.com/osbuild/ci/osbuild-composer/-/jobs/1397504979#L2423)

The `api.sh` script for commit 6e22e1de64c18299c80e13a7b9a60049b16ea040 is clearly the correct (up-to-date) one https://github.com/osbuild/osbuild-composer/blob/6e22e1de64c18299c80e13a7b9a60049b16ea040/test/cases/api.sh#L65

Is CI for main using older versions of the osbuild-composer-tests RPM?

### PR DESCRIPTION
**Pull request by atodorov**
Closes #1527.

### CODE REVIEW

File: schutzbot/update_github_status.sh [hunk 1]
```diff
@@ -1,5 +1,11 @@
 #!/bin/bash
 
+
+# Don't overwrite CI statuses on main branch for nightly pipelines
+if [[ "$CI_PIPELINE_SOURCE" == "schedule" && "$CI_COMMIT_BRANCH" == "main" ]]; then
```
Comment (teg) [inline schutzbot/update_github_status.sh:5]: The same condition should apply to all branches, should it not? Happy to merge this as is, and fix in a follow-up, if that is the case.

**Review by achilleas-k — APPROVED** [review]

**Comment by achilleas-k** [comment]
Approved.  As for #1527, I see how t his would avoid overriding the CI status for main, but isn't it also an issue that tests for a commit run with old test scripts?  Do we generally expect the first nightly to fail when a test script needs to be modified until the nightly builds catch up?

**Comment by atodorov** [comment]
> but isn't it also an issue that tests for a commit run with old test scripts? Do we generally expect the first nightly to fail when a test script needs to be modified until the nightly builds catch up?

That's a separate issue so let's discuss it separately. Feel free to join the Composer QE meeting tomorrow or open a ticket/issue if you can't attend.

File: schutzbot/update_github_status.sh (hunk 1 above)
Comment (atodorov) [reply schutzbot/update_github_status.sh:5]: ATM the nightly pipeline runs only against `main` as a branch. OTOH removing the branch check doesn't seem to have negative effects. If we happen to run the nightly pipeline against a PR or another branch (e.g. when changing the pipeline) then we'll get the statuses from the pipeline related to the PR and they will not be overwritten with statuses from the scheduled execution, which is what we want. I'll push an update.

**Comment by henrywang** [comment]
@atodorov The `ostree-ng.sh` fail is not caused by this PR but [bz#1979272](https://bugzilla.redhat.com/show_bug.cgi?id=1979272). I have a PR #1529 to fix the issue. Could you please rebase after my PR landed? Thanks.

**Comment by atodorov** [comment]
> @atodorov The `ostree-ng.sh` fail is not caused by this PR but [bz#1979272](https://bugzilla.redhat.com/show_bug.cgi?id=1979272). I have a PR #1529 to fix the issue. Could you please rebase after my PR landed? Thanks.

Rebase isn't strictly necessary b/c the two PRs aren't related and b/c this one will still fail due to the missing image types in 8.5. In any case I can't click the merge button on either of the two PRs b/c I don't have sufficient permissions.

**Review by teg — APPROVED** [review]

**Review by ondrejbudai — COMMENTED** [review]
Looks good thanks. Can you please rebase this PR so it doesn't fail on the ostree test?

**Review by ondrejbudai — APPROVED** [review]

**Review by achilleas-k — APPROVED** [review]

**Comment by atodorov** [comment]
I've rebased, ostree tests passed but there are other unrelated failures. I need somebody with more perms to click on the Merge button.

**Comment by ondrejbudai** [comment]
Weird, seems like the CDN content had issues for a while.  I restarted a tests affected by this issue.

**Review by ondrejbudai — APPROVED** [review]

**Pull request merged**